drivers/windows/R/batch.R

```diff
@@ -1,19 +1,48 @@
 write_batch_task_run <- function(task_id, config, path_root) {
+
+  run_on_linux <- config$platform == "linux"
+  template_file <- if (run_on_linux) "task_run.sh" else "task_run.bat"
+  out_script <- if (run_on_linux) SH_RUN else BATCH_RUN
+  write_os_lines <- if (run_on_linux) write_linux_lines else writeLines
+
   data <- template_data_task_run(task_id, config, path_root)
-  str <- glue_whisker(read_template("task_run.bat"), data)
-  path <- path_to_task_file(path_root, task_id, BATCH_RUN)
-  writeLines(str, path)
+  str <- glue_whisker(read_template(template_file), data)
+  path <- path_to_task_file(path_root, task_id, out_script)
+  write_os_lines(str, path)
+
+  if (run_on_linux) {
+    path_dat <- prepare_path(path, config$shares)
+    linux_path <- path_on_linux(path_dat)
+    wrap_path <- path_to_task_file(path_root, task_id, SH_WRAP_RUN)
+    write_linux_lines(
+      sprintf("python -u /opt/hpcnodemanager/kwrap.py %s", linux_path),
+        wrap_path)
```

Comment:
Can you add a comment explaining what any of this does please?

Reply:
Comments added in batch-linux.R